### DISCUSSION TITLE
How to place an order on-chain?

### DISCUSSION THREAD

**Discussion by oes-varok** [ProjectOpenSea/seaport-js#177] [Q&A] [answered]
Everything is so convoluted here.
I saw a post on the main seaport repo that says that i should call 'validate()' on the seller side to place the order on-chain. But calling validate() from seaport-js doesn't seem to do anything? I don't get a use-case to execute unlike 'createOrder()' for example.

```
      const { executeAllActions } = await this._seaportjs.createOrder(
      {
          offer: [
          {
              itemType: ItemType.ERC721, // create erc721 order
              token: this._token.address, // with our token
              identifier: tokenIdStr // tokenIdStr, // and a specified id
          },
          ],
          consideration: [
          {
              amount: ethers.utils.parseEther(ethamtstr).toString(), // desired payment amount
              recipient: this.state.selectedAddress, // the payment goes back to the user creating this order
          },
          ],
      },
      this.state.selectedAddress
      );

      const order = await executeAllActions();
      console.log(order);

      this.setState({ lastOrder: order} );

      
      const { executeValidate } = this._seaportjs.validate(
        [order],
        this.state.selectedAddress
      );

      // const validation = await executeValidate();
      console.log(executeValidate);
```

Am I completely wrong in understanding this? Does seaport not place the orders on-chain? I have to keep the seller's order in some off-chain database so buyers can use it to call fulfillOrder()?

**Comment by ryanio** [comment, marked as answer]
you can post the order on chain with validate and if it passes all validation it will appear on opensea

validate returns a transaction so you have to await it, like this:
```js
const validateTx = this._seaportjs.validate(
        [order],
        this.state.selectedAddress
      );
const result = await validateTx();
console.log(result);
```

but you can also get an opensea api key and send using [opensea-js](https://github.com/ProjectOpenSea/opensea-js) and skip the gas fee
